### DISCUSSION TITLE
Question: Build and run WinGet CLI as stand-alone application?

### DISCUSSION THREAD

**Discussion by vikekh** [microsoft/winget-cli#1150] [Q&A] [answered]
I want to try to build and run WinGet CLI as a stand-alone application, as I want to use it via Ansible and currently I'm having issues with `AppExecLink`.

I downloaded the repo and built for x64. I see an `src\x64\Release\AppInstallerCLI.exe` built, but I figure this isn't a stand-alone application I can rename and put in my path, right? Very inexperienced with desktop application development and specifically in C++.

Edit: I wasn't aware of the discussions tab!

**Comment by JohnMcPMS** [comment, marked as answer]
This isn't an officially supported path, but we have attempted to keep it possible.

The only files you should need are the .exe and the .pri file that lands next to it.  However, you need some minimum Windows version for the Windows code to find the .pri file if it is just loose next to it (but I don't know what that version is).

If you have the VC runtime installed, this .exe (with .pri) should be able to run standalone just as the packaged `winget` would.  It does use a different storage location for settings and files, and isn't as secure against certain kinds of attacks on the supply chain (since it doesn't deploy the index data, nor does it verify the integrity of the index data).  So I wouldn't suggest using it anywhere that security is critical.  These are not insurmountable issues for the code, but they would take time to implement.  I believe that there are several existing issues if you want to 👍 the request to officially support a non-MSIX package distribution.

**Reply by vikekh** [reply]
Thank you!
